### PR TITLE
Add PHPstan and move code quality tools to subfolder

### PR DESCRIPTION
**Pull request by bezin**
Adds basic PHPstan configuration to the skeleton bundle

### CODE REVIEW

**Comment by leofeyer** [comment]
If we keep adding CI related stuff, we should probably promote using `bamarni/composer-bin-plugin` and move PHPStan and ECS into a `tools` subfolder (as we do in `contao/contao`).

**Comment by bezin** [comment]
I moved ECS and PHPstan to the tools subfolder and use the `composer-bin-plugin`. Like this setup a lot.

**Comment by bytehead** [comment]
What about the CI tasks for PHPStan and CS?

**Comment by bezin** [comment]
> What about the CI tasks for PHPStan and CS?

My main goal was to have these tools pre-configured at hand on the CLI. I wonder if we should add GitHub specific stuff? I mean, sure, everything else is already opinionated, too, but then, I use Bitbucket to host private bundles I develop for clients 🤔

**Comment by bytehead** [comment]
I was more thinking about the bundle itself to have PHPStan and CS checks in the future. :)

**Comment by bezin** [comment]
Aye, that makes sense 👍

**Comment by bezin** [comment]
Would be nice if somebody could check if the workflow file is alright. I only copied it from the monorepo and adjusted it to my best knowledge while having zero experience with Github workflows😉

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+on:
+    pull_request: ~
+    push:
+        branches:
+            - main
+        tags:
+            - '*'
+    schedule:
+        - cron: 0 13 * * MON,THU
+
```
Comment (ausi) [inline .github/workflows/ci.yml]:
```suggestion
on:
    pull_request: ~
    push: ~
```

`schedule` is very specific IMO, triggering on every push and on every pull request seems like a good default to me.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+jobs:
+    ecs:
```
Comment (ausi) [inline .github/workflows/ci.yml:8]: We should also add a job that runs the unit tests for every supported PHP version I think.

Comment (leofeyer) [reply .github/workflows/ci.yml:8]: ECS should be run with the lowest supported PHP version only, so there will be no incompatible changes.

File: .gitignore [hunk 3]
```diff
@@ -5,3 +5,7 @@
 # PhpUnit
 /.phpunit.result.cache
 /phpunit.xml
+
+# Tools
+/tools/*/composer.lock
+/tools/*/vendor
```
Comment (leofeyer) [inline .gitignore]:
```suggestion
/tools/*/vendor
```

The composer.lock file should not be excluded, especially when the bundle is reusable.

Comment (bezin) [reply .gitignore]: See [09a11ec](https://github.com/contao/skeleton-bundle/pull/11/commits/09a11eccc04f15ca972c207eb94c50c0a8d0e406)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bezin) [reply .github/workflows/ci.yml]: See [51bc1b9](https://github.com/contao/skeleton-bundle/pull/11/commits/51bc1b9f6fc4f13ca7a865caec4ac807716668bd)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (bezin) [reply .github/workflows/ci.yml:8]: Ok, I think these are two different things: 

* ECS: Will update ECS to run with PHP 7.4. What about PHPStan? Newest PHP version or lowest supported?
* Unit tests: I copied them from the monorepo as well, but thought it was over the top. On the other hand: What's the harm? Will add them, too.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+on:
+    pull_request: ~
+    push: ~
+
+jobs:
+    ecs:
+        name: ECS
+        runs-on: ubuntu-latest
+        if: github.event_name != 'push'
```
Comment (ausi) [inline .github/workflows/ci.yml]: I’d remove this check (is specific to contao/contao IMO)

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                    composer install --no-interaction --no-progress
+                    composer bin phpstan install --no-interaction --no-progress
+
+            -   name: Run PHPStan
+                run: tools/phpstan/vendor/bin/phpstan analyse --no-progress
+
+    tests:
+        name: PHP ${{ matrix.php }}
+        runs-on: ubuntu-latest
+        if: github.event_name != 'push'
```
Comment (ausi) [inline .github/workflows/ci.yml]: Same here :)

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            -   name: Install the dependencies
+                run: composer install --no-interaction --no-progress
+
+            -   name: Run the unit tests
+                run: vendor/bin/phpunit --colors=always
+
+    nightly:
+        name: PHP 8.2
+        runs-on: ubuntu-latest
+        if: github.event_name != 'push'
```
Comment (ausi) [inline .github/workflows/ci.yml]: :)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ausi) [reply .github/workflows/ci.yml:8]: For phpstan usually the newest I think :)

**Review by ausi — APPROVED** [review]
❤️

**Comment by leofeyer** [comment]
Somehow Github actions does not work: https://github.com/contao/skeleton-bundle/pull/11/checks 🤔

**Comment by bezin** [comment]
Maybe it needs to be merged first?

**Comment by leofeyer** [comment]
I don‘t think so. The workflow is executed, but the parser apparently does not find any jobs.

**Comment by leofeyer** [comment]
Ah, it seems that the workflow was executed in your repo. There is also an error message there: https://github.com/bezin/skeleton-bundle/actions/runs/2466969485

**Comment by bezin** [comment]
I debugged and fixed it, but now I am confused more than before: I fixed it by skipping PhpUnitExpectationFixer as described in [this issue](https://github.com/symplify/symplify/issues/3130). But why is this not failing in the monorepo, even though we require the same version of contao/easy-coding-standards? 

Did some more research: Because in the monorepo, even though 4.13 only requires PHP 7.4, [we run ECS with PHP 8.1](https://github.com/contao/contao/blob/4.13/.github/workflows/ci.yml). Shouldn't this be running with 7.4. too?

File: ecs.php [hunk 7]
```diff
@@ -6,11 +6,17 @@
...
-    $ecsConfig->sets([__DIR__.'/vendor/contao/easy-coding-standard/config/contao.php']);
+    $ecsConfig->sets([__DIR__.'/tools/ecs/vendor/contao/easy-coding-standard/config/contao.php']);
 
     $ecsConfig->ruleWithConfiguration(HeaderCommentFixer::class, [
         'header' => "This file is part of [package name].\n\n(c) John Doe\n\n@license LGPL-3.0-or-later"
     ]);
 
+    if (PHP_VERSION_ID < 80000) {
+        $ecsConfig->ruleWithConfiguration(\PhpCsFixer\Fixer\ControlStructure\TrailingCommaInMultilineFixer::class, ['elements' => ['arrays'], 'after_heredoc' => true]);
+        // SKip PhpUnitExpecationFixer (see https://github.com/symplify/symplify/issues/3130)
```
Comment (bytehead) [inline ecs.php]:
```suggestion
        // Skip PhpUnitExpecationFixer (see https://github.com/symplify/symplify/issues/3130)
```

Typo 😎

**Review by bytehead — CHANGES_REQUESTED** [review]

**Review by bytehead — APPROVED** [review]
❤️

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bezin.

**Pull request merged**